### DISCUSSION TITLE
Using dunfell 32.5.0

### DISCUSSION THREAD

**Discussion by tirankas** [OE4T/meta-tegra#598] [Q&A] [answered]
Hello,
I am trying to build the new jetpack 4.5 using the dunfell 32.5.0 branch fr meta tegra.
I don't understand the dependency on the other external layers like poky, virtualization, openembedded etc.
In the previous dunfell version these meta layers had a dedicated tag so everything was aligned, I checked out the same branch on each of these layers.
But for the new branch (32.5.0) there is no such tag on these layers and every tag/master I select j get incompatibility issues.

Is there a known setup that should work out of the box or I need to manually update compatibilities in each layer?

Thanks

**Comment by madisongh** [comment]
> In the previous dunfell version these meta layers had a dedicated tag so everything was aligned

I'm not sure what you mean by this, but as the branch name implies, you should use the `dunfell` branch for any other layers you include. We do have a [demo distro](https://github.com/OE4T/tegra-demo-distro) setup that includes a bunch of layers as git submodules.  It has a `dunfell-l4t-r32.5.0` branch with all of the layers set appropriately, and it gets updated fairly often.

**Reply by tirankas** [reply]
Thanks for the prompt reply.
I am picking the meta-tegra from dunfell-l4t-r32.5.0 branch.
I still need to clone the following:
git clone -b ${BRANCH} git://git.yoctoproject.org/poky.git poky-${BRANCH}
git clone https://git.yoctoproject.org/git/meta-virtualization
git clone https://git.openembedded.org/meta-openembedded

These three layers have no dunfell-l4t-r32.5.0 branch.
Should I clone these repos with the dunfell branch (and not the dunfell-l4t-r32.5.0)?

Thanks

**Reply by ichergui** [reply]
Hey @tirankas 

Please check [here](https://github.com/OE4T/tegra-demo-distro/tree/dunfell-l4t-r32.5.0/repos)
you can find all repo you need

**Reply by madisongh** [reply]
To answer your question:
> Should I clone these repos with the dunfell branch (and not the dunfell-l4t-r32.5.0)?

Yes.

**Comment by ichergui** [comment, marked as answer]
> Hello,
> I am trying to build the new jetpack 4.5 using the dunfell 32.5.0 branch fr meta tegra.
> I don't understand the dependency on the other external layers like poky, virtualization, openembedded etc.
> In the previous dunfell version these meta layers had a dedicated tag so everything was aligned, I checked out the same branch on each of these layers.
> But for the new branch (32.5.0) there is no such tag on these layers and every tag/master I select j get incompatibility issues.
> 
> Is there a known setup that should work out of the box or I need to manually update compatibilities in each layer?
> 
> Thanks

As @madisongh suggest, please use the [tegra-demo-distro repo](https://github.com/OE4T/tegra-demo-distro) 
Let us know if you need help.

**Reply by tirankas** [reply]
Thanks guys, this really helped out!
I tried with the tegra demo distro as suggested and I see this:
. ./setup-env --machine jetson-nano-devkit
Error: TEMPLATECONF value points to non existence directory meta-poky/conf

I will try the usual way of manually cloning repos and build.

Thanks

**Reply by ichergui** [reply]
Please try with the following commands

```
$ git clone https://github.com/OE4T/tegra-demo-distro.git
$ cd tegra-demo-distro
$ git checkout origin/dunfell-l4t-r32.5.0 -b dunfell-l4t-r32.5.0
$ git submodule update --init --recursive 
$ unset NVIDIA_DEVNET_MIRROR                                    
$ export NVIDIA_DEVNET_MIRROR="/PATH/TO/JETSON_NANO/nvidia/sdkm_downloads"
$ . ./setup-env --machine jetson-nano-devkit --distro tegrademo build_jetson_nano
```
You should see the following result:
```
You had no conf/local.conf file. This configuration file has therefore been
created for you with some default values. You may wish to edit it to, for
example, select a different MACHINE (target hardware). See conf/local.conf
for more information as common configuration options are commented.

You had no conf/bblayers.conf file. This configuration file has therefore been
created for you with some default values. To add additional metadata layers
into your configuration please add entries to conf/bblayers.conf.

The Yocto Project has extensive documentation about OE including a reference
manual which can be found at:
    http://yoctoproject.org/documentation

For more information about OpenEmbedded see their website:
    http://www.openembedded.org/

### Shell environment set up for builds. ###

You can now run 'bitbake <target>'
Targets for building Tegra demo images with test applications:
    core-image-minimal  - minimally bootable image (no demo apps, no graphics)
    demo-image-base     - basic image with no graphics
    demo-image-egl      - basic image with DRM/EGL, no window manager
    demo-image-sato     - X11 image with 'sato' UI
    demo-image-weston   - Wayland with Weston compositor
    demo-image-full     - X11/sato UI plus docker, openCV, multimedia API samples

Your build environment has been configured with:
    MACHINE=jetson-nano-devkit
    DISTRO=tegrademo
    NVIDIA_DEVNET_MIRROR=file:///PATH/TO/JETSON_NANO/nvidia/sdkm_downloads"
```

After this, you can trig a build .e.g. `$ bitbake demo-image-base`

**Reply by tirankas** [reply]
Thanks, I ended up doing something similar which should work as your suggestion.
I am a bit confused now. The tegra dunfell 32.5.0 suppose to support the jetpack 4.5 but from the last reply I see that NVIDIA_DEVNET_MIRROR points to jetpack_4.4.1.
Is 4.5 supported or 4.4.1?

Thanks

**Reply by ichergui** [reply]
My bad
It was copy/paste issue.
You should use `Jetpack 4.5` for `tegra-demo-distro` repo - branch `dunfell-l4t-r32.5.0`.

**Reply by ichergui** [reply]
Hey @tirankas 

Could you please give us updates about your issue ?

Hopefully, all good in your side
